x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/TranslationAware.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.core.expression;
+
+import org.elasticsearch.xpack.esql.core.planner.TranslatorHandler;
+import org.elasticsearch.xpack.esql.core.querydsl.query.Query;
+
+/**
+ * Expressions can implement this interface to control how they would be translated and pushed down as Lucene queries.
+ * When an expression implements {@link TranslationAware}, we call {@link #asQuery(TranslatorHandler)} to get the
+ * {@link Query} translation, instead of relying on the registered translators from EsqlExpressionTranslators.
+ */
+public interface TranslationAware {
+    Query asQuery(TranslatorHandler translatorHandler);
```

Comment:
I think it would make sense to centralize in this interface all the pushdown logic, including
- capabilities, eg. can it be pushed down to Lucene at all? Maybe sometimes it depends on its inputs?
- constraints, eg. does this function need to be pushed down to be executed? Does it also have an inline implementation that works without pushdown?

The first point is covered by [PushFiltersToSource](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/physical/local/PushFiltersToSource.java#L255-L259) physical rule. I'm not sure if we want ad-hoc logic for each function or if we can simplify it to a single and generic behavior (ie. removing the `if (exp instanceof Term`). In both cases I'd rather have an `exp instanceof TranslationAware` there, so that one day we can also let other operators/functions implement it and remove that long list of `instanceof`s.

For the second point I'm thinking about the [Verifier logic](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java#L833) around FullTextFunctions. It's a complex topic that involves some deep understanding of how commands interact with each other, I put it here for completeness but probably we can address it in a follow-up PR